packages/ra-core/src/sideEffect/saga/cancelSaga.js

```diff
@@ -0,0 +1,119 @@
+import { all, call, take, takeEvery, put, race } from 'redux-saga/effects';
+import { delay } from 'redux-saga';
+import { push } from 'react-router-redux';
+
+import {
+    CRUD_UPDATE,
+    CRUD_UPDATE_MANY,
+    CRUD_DELETE,
+    CRUD_DELETE_MANY,
+} from '../../actions/dataActions';
+import {
+    showNotification,
+    hideNotification,
+} from '../../actions/notificationActions';
+import {
+    CANCELLABLE,
+    CANCEL,
+    startOptimisticMode,
+    stopOptimisticMode,
+} from '../../actions/cancelActions';
+import { refreshView } from '../../actions/uiActions';
+import resolveRedirectTo from '../../util/resolveRedirectTo';
+import { setListSelectedIds } from '../../actions/listActions';
+
+function* handleCancelRace(cancellableAction) {
+    const { payload: { action, delay: cancelDelay } } = cancellableAction;
+    // display cancellable notification
+    switch (action.type) {
```

Comment:
instead of doing a check on the action type, the notification should be in the action. That would make the saga very generic (and reusable for user actions).